lib/strong_migrations/adapters/abstract_adapter.rb

```diff
@@ -47,7 +47,15 @@ def select_all(statement)
       end
 
       def target_version(target_version)
-        target_version ||= StrongMigrations.target_version
+        target_version ||=
+          if StrongMigrations.target_version.is_a?(Hash)
+            StrongMigrations.target_version.fetch(db_config_name) do
+              raise "StrongMigrations.target_version is not configured for :#{db_config_name}"
+            end
+          else
+            StrongMigrations.target_version
+          end
+
         version =
           if target_version && StrongMigrations.developer_env?
             target_version.to_s
```

Comment:
I think it's better to put all the new logic here so it's not executed unless it'll be used.